libmscore/measure.cpp

```diff
@@ -884,6 +884,30 @@ void Measure::add(Element* e)
                   _mmRest = toMeasure(e);
                   break;
 
+            case ElementType::STAFFTYPE_CHANGE:
+                  {
+                  StaffTypeChange* stc = toStaffTypeChange(e);
+                  Staff* staff = stc->staff();
+                  const StaffType* st = stc->staffType();
+                  StaffType* nst;
+                  // st needs to point to actual stafftype within the stafftypelist for the staff
+                  if (st) {
+                        // setStaffType add this list and returns pointer to that element within list
+                        // we won't need the original after that
+                        nst = staff->setStaffType(tick(), *st);
+                        delete st;
```

Comment:
AddressSanitizer [complains](https://travis-ci.org/musescore/MuseScore/jobs/521757461#L4203) that in one test the StaffType deleted here actually belongs to a staff as it was already added there in `StaffTypeChange::read()`:
https://github.com/musescore/MuseScore/blob/b2bba97a7d4ff1daf5a0baeafc5f9ecf58d23bc3/libmscore/stafftypechange.cpp#L63-L66

As an option to fix it, maybe `StaffTypeChange::read()` should not add a staff type to staff since it is handled by `Measure::add()` after this patch.

Reply:
Huh, I had not considered that.  Thanks, I will walk through this a bit more and hopefully push another update later.